test/unix/main.ml

```diff
@@ -257,18 +259,99 @@ module Readonly = struct
     let ro = clone ~readonly:true () in
     let k1, v1 = (Key.v (), Value.v ()) in
     Index.replace rw k1 v1;
+    Index.ro_sync ro;
     Alcotest.check_raises "Index readonly cannot read if data is not flushed."
       Not_found (fun () -> ignore_value (Index.find ro k1));
     Index.close rw;
     Index.close ro
 
+  let readonly_v_in_sync () =
+    let Context.{ rw; clone; _ } = Context.full_index () in
+    let k = Key.v () in
+    let v = Value.v () in
+    Index.replace rw k v;
+    Index.flush rw;
+    let ro = clone ~readonly:true () in
+    check_index_entry ro k v;
+    Index.close rw;
+    Index.close ro
+
+  let _readonly_add_log_before_clear () =
+    let Context.{ rw; clone; _ } = Context.empty_index () in
+    let ro = clone ~readonly:true () in
+    let k1, v1 = (Key.v (), Value.v ()) in
+    Index.replace rw k1 v1;
+    Index.flush rw;
+    Index.ro_sync ro;
+    check_index_entry ro k1 v1;
+    Index.clear rw;
+    Index.ro_sync ro;
+    Alcotest.check_raises (Printf.sprintf "Found %s key after clearing." k1)
+      Not_found (fun () -> ignore_value (Index.find ro k1));
+    Index.close rw;
+    Index.close ro
+
+  let _readonly_add_index_before_clear () =
+    let Context.{ rw; clone; _ } = Context.full_index () in
+    let ro = clone ~readonly:true () in
+    Index.clear rw;
+    let k1, v1 = (Key.v (), Value.v ()) in
+    Index.replace rw k1 v1;
+    for _ = 1 to 5 do
+      let k = Key.v () in
+      let v = Value.v () in
+      Index.replace rw k v
+    done;
+    Index.flush rw;
+    Index.ro_sync ro;
+    check_index_entry ro k1 v1;
+    Index.clear rw;
+    Index.ro_sync ro;
+    Alcotest.check_raises (Printf.sprintf "Found %s key after clearing." k1)
+      Not_found (fun () -> ignore_value (Index.find ro k1));
+    Index.close rw;
+    Index.close ro
+
+  let _readonly_add_after_clear () =
+    let Context.{ rw; clone; _ } = Context.empty_index () in
+    let ro = clone ~readonly:true () in
+    let k1, v1 = (Key.v (), Value.v ()) in
+    Index.replace rw k1 v1;
+    Index.flush rw;
+    Index.ro_sync ro;
+    check_index_entry ro k1 v1;
+    Index.clear rw;
+    let k2, v2 = (Key.v (), Value.v ()) in
+    Index.replace rw k2 v2;
+    Index.flush rw;
+    Index.ro_sync ro;
+    check_index_entry ro k2 v2
```

Comment:
Missing `close`s here.
And perhaps that could check that `k1, v1` can't be found anymore?